pkg/registry/core/rest/storage_core.go

```diff
@@ -223,7 +257,7 @@ func (p *legacyProvider) NewRESTStorage(apiResourceConfigSource serverstorage.AP
 			utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenPodNodeInfo) {
 			nodeGetter = nodeStorage.Node.Store
 		}
-		serviceAccountStorage, err = serviceaccountstore.NewREST(restOptionsGetter, p.ServiceAccountIssuer, p.APIAudiences, p.ServiceAccountMaxExpiration, podStorage.Pod.Store, storage["secrets"].(rest.Getter), nodeGetter, p.ExtendExpiration, p.MaxExtendedExpiration)
+		serviceAccountStorage, err = p.getServiceAccountREST(restOptionsGetter, p.ServiceAccountIssuer, p.APIAudiences, p.ServiceAccountMaxExpiration, podStorage.Pod.Store, storage["secrets"].(rest.Getter), nodeGetter, p.ExtendExpiration, p.MaxExtendedExpiration)
```

Comment:
@serathius was this single call-site actually getting called multiple times when setting up a single kube-apiserver? That would be really surprising to me, and would mean we'd also have to protect every other storage creation in legacyProvider#NewRESTStorage

Reply:
so I don't think this serviceAccountOnce in this PR is actually doing anything to prevent double-initialization